proto/ibc/applications/interchain_accounts/v1/types.proto

```diff
@@ -25,3 +20,9 @@ message InterchainAccountPacketData {
   bytes data = 2;
   string memo = 3;
 }
+
+// CosmosTx contains a list of sdk.Msg's. It should be used when sending transactions to an SDK host chain.
+message CosmosTx {
```

Comment:
When trying to embed repeated `Any`s into the Packet Data type, I realized JSON marshalling will fail since the proto json codec will try to unpack the Any into its interface using the UnpackInterfaces implementation. This would require specifying what type of Any the messages are (ie sdk.Msg)

Requiring the controller chain to marshal the packet data also requires the controller chain to register every type sent to the host chain on its codec. This may become a large burden over time

Instead, I think the better solution is to push serialization logic onto the ica-auth module. This means each ica-auth module will need to be able to marshal the types they would like to send to the host chain. They pass in the ICA PacketData in the API of TrySendTx. The controller chain does very basic validation of this packet data and then passes this along after some basic checks

This prevents the controller chain from needing to register every type each host chain it is talking to has registered. Each ica-auth module only needs to register the types it uses and we will expose a serialization function ica-auth modules can all to get the data for the ica packet data

CosmosTx will only be used by ica-auth modules sending to SDK chains. The controller code should not need to understand this type (except to expose the helper function to serialize a CosmosTx)

Reply:
I like this approach & I think the idea to expose the deserialize/serial fns to the `ica-auth` module makes for a clean API. Nice work :+1: